src/main/java/gyro/azure/keyvault/VaultResource.java

```diff
@@ -0,0 +1,476 @@
+ *         enable-disk-encryption: true
+ *         enable-soft-delete: false
+ *
+ *         access-policy
+ *             key-permissions: ["get", "import", "list", "delete"]
+ *             certificate-permissions: ['get', 'update', 'create', 'list', 'delete', 'import', 'backup', 'recover', 'restore', 'setissuers', 'deleteissuers', 'purge', 'listissuers', 'getissuers', 'managecontacts', 'manageissuers']
+ *             secret-permissions: ["get"]
+ *             object-id: "e0afa33f-9882-4cdc-abb8-c613a8949f9e"
+ *         end
+ *
+ *         access-policy
+ *             key-permissions: ["get", "import", "list", "delete"]
+ *             certificate-permissions: ['get', 'update', 'create', 'list', 'delete', 'import', 'backup', 'recover', 'restore', 'setissuers', 'deleteissuers', 'purge', 'listissuers', 'getissuers', 'managecontacts', 'manageissuers']
+ *             secret-permissions: ["get"]
+ *             object-id: "b7d674a2-7e88-47af-b772-5d90b4bf965a"
+ *         end
+ *
+ *         tags: {
+ *             Name: "key-vault-examples"
+ *         }
+ *     end
+ *
+ * Certificate Commands
+ * --------------------
+ *
+ * A set of commands that lets you manage certificates (``add``, ``remove`` or ``list``) of a vault that you are managing using ``gyro``. In order to use these commands, make sure to add proper access policy to the vault for the service principal you are using for the ``azure-provider``.
+ *
+ * The following commands are supported:
+ *
+ * **Add Certificate**
+ *
+ * Adds a certificate to a vault using your certificate file (.pfx). Access policy needed (Upload, Insert).
+ *
+ * .. code::
+ *
+ *     gyro azure vault add-certificate <vault-name> <cert-name> <path> --password <password>
+ *
+ * - ``vault-name`` - The name of the vault resource defined in your config where you want to create your certificate.
+ * - ``cert-name`` - The name of the certificate that you want to create when you import the certificate file.
+ * - ``cert-path`` - The path pointing to the certificate file to be uploaded. Only ``.pfx`` files are supported.
+ * - ``password`` - An optional password if the certificate file was encrypted with one.
+ *
+ * **Remove Certificate**
+ *
+ * Remove a certificate from the vault. Access policy needed (delete).
+ *
+ * .. code::
+ *
+ *     gyro azure vault remove-certificate <vault-name> <cert-name>
+ *
+ * - ``vault-name`` - The name of the vault resource defined in your config from which to remove the certificate.
+ * - ``cert-name`` - The name of the certificate that you want to remove.
+ *
+ * **List Certificate**
+ *
+ * List certificates of a vault. Access policy needed (List).
+ *
+ * .. code::
+ *
+ *     gyro azure vault list-certificate <vault-name>
+ *
+ * - ``vault-name`` - The name of the vault resource defined in your config that you want to list certificates from.
+ * - ``show-thumbprint`` - An option that shows the x509 thumbprint of the certificate.
+ *
+ *
+
+ */
+@Type("vault")
```

Comment:
This should be `@Type("key-vault")`. Update any examples/docs as well. Might also rename all the classes to be consistent.

Reply:
Updated and renamed class to append with `key` as well.